source/pkger/pkger.go

```diff
@@ -0,0 +1,83 @@
+package pkger
+
+import (
+	"fmt"
+	"net/http"
+	stdurl "net/url"
+
+	"github.com/golang-migrate/migrate/v4/source"
+	"github.com/golang-migrate/migrate/v4/source/httpfs"
+	"github.com/markbates/pkger"
+	"github.com/markbates/pkger/pkging"
+)
+
+func init() {
+	source.Register("pkger", &Pkger{})
+}
+
+// Pkger is a source.Driver that reads migrations from instances of
+// pkging.Pkger.
+type Pkger struct {
+	httpfs.PartialDriver
+}
+
+// Open implements source.Driver. The path component of url will be used as the
+// relative location of migrations. The returned driver will use the package
+// scoped pkger.Open to access migrations.  The relative root and any
+// migrations must be added to the global pkger.Pkger instance by calling
+// pkger.Apply. Refer to Pkger documentation for more information.
+func (p *Pkger) Open(url string) (source.Driver, error) {
+	u, err := stdurl.Parse(url)
+	if err != nil {
+		return nil, err
+	}
+
+	// wrap pkger to implement http.FileSystem.
+	fs := fsFunc(func(name string) (http.File, error) {
```

Comment:
Should be able to refactor this to return `WithInstance(pkger, u.Path)`. Based on the [pkgr.Open()](https://github.com/markbates/pkger/blob/v0.15.1/pkger.go#L34), we'd probably have to use [here.Current()](https://godoc.org/github.com/gobuffalo/here#Here.Current).

This is not a blocker for this PR since the difference in complexity is arugable...
e.g. the setup required to use `WithInstance()` may not be worth the duplicated code

Lemme know if you don't think this refactor is worth it and I'll go ahead and merge the PR as is.

Reply:
I say go ahead and merge it as is. I'm not 100% sure, but if I change my mind I'll go ahead and contribute a refactor :)

Thanks!